### PR TITLE
Create comparable tab controller examples in new structure

### LINKED ISSUE #830: Example: tab-controller
Split appropriate existing examples into separate files using functional approach and icache and move into the correct `/src/examples/widgets` subdirectory.

### PR DESCRIPTION
**Pull request by maier49**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Resolves #830

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* dojo.widgets: https://dojowidgets-5waolwdu8.now.sh

File: src/examples/src/widgets/tab-controller/ButtonAlignment.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import TabController, { Align } from '@dojo/widgets/tab-controller';
+import Tab from '@dojo/widgets/tab';
+import Select from '@dojo/widgets/select';
+
+const factory = create({ icache });
+
+export default factory(function ButtonAlignment({ middleware: { icache } }) {
+	const activeIndex = icache.getOrSet('active', 0);
+	const alignButtons = icache.getOrSet<Align>('align', Align.top);
```
Comment (agubler) [inline src/examples/src/widgets/tab-controller/ButtonAlignment.tsx]: Does this need the generic?

File: src/examples/src/widgets/tab-controller/Closeable.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { tsx, create } from '@dojo/framework/core/vdom';
+import icache from '@dojo/framework/core/middleware/icache';
+import { includes } from '@dojo/framework/shim/array';
+
+import TabController from '@dojo/widgets/tab-controller';
+import Tab from '@dojo/widgets/tab';
+
+const factory = create({ icache });
+
+export default factory(function Basic({ middleware: { icache } }) {
```
Comment (agubler) [inline src/examples/src/widgets/tab-controller/Closeable.tsx]: `function Closeable`

File: src/examples/src/widgets/tab-controller/Disabled.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import Tab from '@dojo/widgets/tab';
 
 const factory = create({ icache });
 
-export default factory(function Basic({ middleware: { icache } }) {
+export default factory(function Closeable({ middleware: { icache } }) {
```
Comment (agubler) [inline src/examples/src/widgets/tab-controller/Disabled.tsx]: `Disabled`?

**Comment by maier49** [comment]
@agubler This is up to date now and I believe I've addressed all your feedback

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-7ccrkvdd7.now.sh
* Tests Deployment: https://dojowidgets-eoi5qvxyr.now.sh

**Pull request merged**